devito/types/parallel.py

```diff
@@ -252,7 +252,7 @@ class DeviceID(DeviceSymbol):
 
     @property
     def default_value(self):
-        return -1
+        return configuration['deviceid']
```

Comment:
We can return a `default_value` based on global state

I think the better way to go here would be overriding `_arg_values` (from `Scalar`)

Within the override, we check, in order:

* whether kwargs has a user-override for `deviceid`
* whether `configuration['deviceid']` is != `default_value`
* else return `default_value`